File: backend/src/main/java/com/fretcorridor/backend/shipment/entity/Offer.java
package com.fretcorridor.backend.shipment.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

/**
 * Offre proposee pour une demande (UC-MKT-02). Generee par
 * MockOfferGenerationService en V1 - voir sa Javadoc pour le detail de la
 * simplification assumee (pas de vrai moteur de matching dans ce perimetre).
 */
@Entity
@Table(name = "offers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Offer {

    @Id
    @GeneratedValue
    private UUID id;

    @Column(name = "shipment_request_id", nullable = false)
    private UUID shipmentRequestId;

    @Column(name = "carrier_display_name", nullable = false, length = 150)
    private String carrierDisplayName;

    @Column(name = "carrier_rating", nullable = false, precision = 2, scale = 1)
    private BigDecimal carrierRating;

    @Column(name = "price_xaf", nullable = false, precision = 12, scale = 2)
    private BigDecimal priceXaf;

    @Column(name = "estimated_pickup_at", nullable = false)
    private Instant estimatedPickupAt;

    @Column(name = "estimated_delivery_at", nullable = false)
    private Instant estimatedDeliveryAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private OfferStatus status = OfferStatus.PROPOSEE;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = Instant.now();
    }
}

File: mobile/lib/features/offers/data/models/offer.dart
/// Offre proposee pour une demande (correspond au DTO OfferResponse).
class Offer {
  Offer({
    required this.id,
    required this.shipmentRequestId,
    required this.carrierDisplayName,
    required this.carrierRating,
    required this.priceXaf,
    required this.estimatedPickupAt,
    required this.estimatedDeliveryAt,
    required this.status,
  });

  final String id;
  final String shipmentRequestId;
  final String carrierDisplayName;
  final double carrierRating;
  final double priceXaf;
  final DateTime estimatedPickupAt;
  final DateTime estimatedDeliveryAt;
  final String status; // PROPOSEE, ACCEPTEE, REFUSEE, EXPIREE

  factory Offer.fromJson(Map<String, dynamic> json) => Offer(
        id: json['id'],
        shipmentRequestId: json['shipmentRequestId'],
        carrierDisplayName: json['carrierDisplayName'],
        carrierRating: (json['carrierRating'] as num).toDouble(),
        priceXaf: (json['priceXaf'] as num).toDouble(),
        estimatedPickupAt: DateTime.parse(json['estimatedPickupAt']),
        estimatedDeliveryAt: DateTime.parse(json['estimatedDeliveryAt']),
        status: json['status'],
      );
}
